packages/api/test/e2e/promise-queries.spec.ts

```diff
@@ -177,20 +177,197 @@ describe.skip('Promise e2e queries', () => {
     );
   });
 
-  it('queries state using double map key', async () => {
-    // TODO Update ['any', '0x1234'] to the key of a known event topic and update '[]' to the expected value
-    const eventTopics = await api.query.system.eventTopics(['any', '0x1234']);
+  describe('with plain type', async () => {
+    const TOTAL_ISSUANCE = 12582912;
+    it('queries correct value', async () => {
+      const totalIssuance = await api.query.balances.totalIssuance() as Balance;
 
-    expect(eventTopics.toString()).toEqual('[]');
+      expect(totalIssuance.toNumber()).toEqual(TOTAL_ISSUANCE);
+    });
+
+    it('queries correct value at a specified block', async () => {
+      const header = await api.rpc.chain.getHeader() as Header;
+      const totalIssuanceAt = await api.query.balances.totalIssuance.at(header.hash) as Balance;
+
+      expect(totalIssuanceAt.toNumber()).toEqual(TOTAL_ISSUANCE);
+    });
+
+    it('subscribes to query and get correct result', async (done) => {
+      api.query.balances.totalIssuance((totalIssuance) => {
+        expect(totalIssuance.toNumber()).toEqual(TOTAL_ISSUANCE);
+        done();
+      });
+    });
+
+    it('queries correct hash', async () => {
+      const hash = await api.query.balances.totalIssuance.hash();
+
+      expect(hash).toBeDefined();
+    });
+
+    it('gets correct key', async () => {
+      const key = await api.query.balances.totalIssuance.key();
+      const totalIssuanceData = await api.rpc.state.getStorage(key) as Option<any>;
+      const totalIssuanceRPC = new Balance(totalIssuanceData.unwrapOr(undefined));
+
+      expect(totalIssuanceRPC.toNumber()).toEqual(TOTAL_ISSUANCE);
+    });
+
+    it('queries correct size', async () => {
+      const size = await api.query.balances.totalIssuance.size();
+
+      expect(size.toNumber()).not.toEqual(0);
+    });
   });
 
-  it('subscribes to queries using double map key', async (done) => {
-    return (
-      // TODO Update ['any', '0x1234'] to the key of a known event topic and update '[]' to the expected value
-      api.query.system.eventTopics(['any', '0x1234'], (eventTopics) => {
-        expect(eventTopics.toString()).toEqual('[]');
+  describe('with map type', async () => {
+    it('queries correct value', async () => {
+      const balance = await api.query.balances.freeBalance(keyring.alice.address()) as Balance;
+
+      expect(balance.isZero()).toBe(false);
+    });
+
+    it('queries correct value at a specified block', async () => {
```

Comment:
These test cases assume the account Alice is only used in tests(the balance of Alice won't change between `api.query.balances.freeBalance(keyring.alice.address());` and `api.query.balances.freeBalance.at(header.hash, keyring.alice.address())` ). We can change this to just check the queried balance is greater than 0.